### PR TITLE
added parameter sshd_config_subsystem_sftp_options for https://access…

### PR DESCRIPTION
**Pull request by depohlm**
added parameter sshd_config_subsystem_sftp_options (https://access.redhat.com/articles/1374633)

### CODE REVIEW

File: templates/sshd_config.erb [hunk 1]
```diff
@@ -238,7 +238,7 @@ XAuthLocation <%= @sshd_config_xauth_location_real %>
 
 <% end -%>
 # override default of no subsystems
-Subsystem sftp <%= @sshd_config_subsystem_sftp_real %>
+Subsystem sftp <%= @sshd_config_subsystem_sftp_real %> <%= @sshd_config_subsystem_sftp_options %>
```
Comment (ghoneycutt) [inline templates/sshd_config.erb]: the trailing whitespace is causing an issue with the spec tests. You can enclose it and the part you added in an if statement as shown above.

File: manifests/init.pp [hunk 2]
```diff
@@ -73,6 +73,7 @@
   $sshd_banner_mode                       = '0644',
   $sshd_config_xauth_location             = 'USE_DEFAULTS',
   $sshd_config_subsystem_sftp             = 'USE_DEFAULTS',
+  $sshd_config_subsystem_sftp_options     = undef,
```
Comment (ghoneycutt) [inline manifests/init.pp]: This new parameter needs a spec test associated with it.

**Comment by ghoneycutt** [comment]
Thank you! That should solve the template issue, though we need spec tests added before this can be merged.

File: manifests/init.pp [hunk 3]
```diff
@@ -321,6 +322,10 @@
     $sshd_config_subsystem_sftp_real = $sshd_config_subsystem_sftp
   }
 
+  if $sshd_config_subsystem_sftp_options != undef {
```
Comment (depohlm) [inline manifests/init.pp]: Spec added here.

Comment (ghoneycutt) [reply manifests/init.pp]: That's the validation piece. Check out this commit to see how spec tests were added.

https://github.com/ghoneycutt/puppet-module-ssh/commit/bc4c8a82a64c41c2d146b038cbc3e8d6a4fc932b

**Comment by depohlm** [comment]
Spec test added.

**Comment by Phil-Friderici** [comment]
Closing this PR as it was made for the v3 version of the module which is outdated now.
v4 does allow to set `Subsystem` in sshd_config with the `$ssh::server::subsystem` parameter.

**Pull request closed without merging**